js/components/frame.js

```diff
@@ -337,8 +337,9 @@ class Frame extends ImmutableComponent {
           this.props.frame)
         const key = this.props.frame.get('key')
         const parsedUrl = urlParse(event.url)
+        // don't change url for non-display protocols like mailto
         if (['http:', 'https:', 'about:', 'chrome:', 'chrome-extension:', 'file:',
-             'view-source:'].includes(parsedUrl.protocol)) {
+             'view-source:', 'ftp:', 'data:'].includes(parsedUrl.protocol)) {
```

Comment:
I just threw ftp in there because I noticed it was missing, but do we want to make this a whitelist or a blacklist? At the moment the only protocol we are aware of that causes a problem is mailto, but there could potentially be other security issues related to not changing urls